lib/ws/ws.js

```diff
@@ -9,13 +9,86 @@ var factory = require('./factory');
 var authMethod = 'auth';
 
 /**
+ * race takes an array of promises and is resolved with the first promise
+ * that resolves, or rejects. After the first resolve or rejection, all
+ * future resolutions and rejections will be discarded.
+ * @param {Promise[]} promises
+ * @return {Promise}
+ */
+function race(promises) {
```

Comment:
Then it would seem that we should polyfill that on frontend instead of here. Promises should have a race method [by specification](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/race).